recipes/_master_package.rb

```diff
@@ -52,6 +66,25 @@
     version node['jenkins']['master']['version']
   end
 
+  bash "Modify packaged log rotation directory" do
+    code <<-EOH
+      sed -i "s|/var/log/jenkins|#{node['jenkins']['master']['log_directory']}|g" /etc/logrotate.d/jenkins
+    EOH
+  end
+
+  bash "Modify packaged log rotation file name" do
+    code <<-EOH
+      sed -i "s|access_log|#{node['jenkins']['master']['access_log']}|g"          /etc/logrotate.d/jenkins
+    EOH
+  end
+
+  bash "Modify packaged access log" do
+    code <<-EOH
+      sed -i "/accessLogger/ s|/var/log/jenkins|#{node['jenkins']['master']['log_directory']}|g" /etc/init.d/jenkins
+      sed -i "/accessLogger/ s|access_log|#{node['jenkins']['master']['access_log']}|g"          /etc/init.d/jenkins
+    EOH
+  end
+
```

Comment:
Same as above, you'll want some guards in place to prevent this from running every chef run.